lib/streamlit/connections/snowflake_connection.py

```diff
@@ -199,15 +199,17 @@ def query(
             ),
             wait=wait_fixed(1),
         )
-        @cache_data(
-            show_spinner=show_spinner,
-            ttl=ttl,
-        )
         def _query(sql: str) -> pd.DataFrame:
             cur = self._instance.cursor()
             cur.execute(sql, params=params, **kwargs)
             return cur.fetch_pandas_all()
 
+        _query.__qualname__ = f"{_query.__qualname__}_{self._connection_name}_{ttl}"
```

Comment:
We should add an explanation of why we're doing this, since it is pretty weird.

Reply:
Ah, yeah -- comments added to each file